web/preferences.js

```diff
@@ -82,8 +102,8 @@ var Preferences = {
    */
   reset: function preferencesReset() {
     return this.initializedPromise.then(function() {
-      this.prefs = Object.create(DEFAULT_PREFERENCES);
-      return this._writeToStorage(DEFAULT_PREFERENCES);
+      this.prefs = Object.create(this.defaults);
```

Comment:
It seems that my previous comment causes issues here, since after a `reset`, attempting to to the `set` a new value fails.
I still think that it makes a certain amount of sense freeze the `defaults`, so could we instead change this line to `this.prefs = Object.assign({}, this.defaults);`, to fix this issue?